appservice/src/main/java/nu/yona/server/subscriptions/rest/UserController.java

```diff
@@ -226,10 +227,35 @@ private String determineRequestingUserId(UUID userId, String requestingUserIdStr
 				() -> addUser(Optional.ofNullable(overwriteUserConfirmationCode), user));
 	}
 
+	private static void assertValidDeviceData(PostPutUserDto postPutUser)
+	{
+		if (postPutUser.deviceName == null)
+		{
+			String hint = "If the device name is not provided, the other properties should not be provided either";
+			Require.isNull(postPutUser.deviceOperatingSystemStr,
+					() -> InvalidDataException.extraProperty("deviceOperatingSystem", hint));
+			Require.isNull(postPutUser.deviceAppVersion, () -> InvalidDataException.extraProperty("deviceAppVersion", hint));
+			Require.isNull(postPutUser.deviceAppVersionCode,
+					() -> InvalidDataException.extraProperty("deviceAppVersionCode", hint));
+		}
+		else
+		{
+			String hint = "If the device name is provided, the other properties must be present too";
+			Require.isNonNull(postPutUser.deviceOperatingSystemStr,
+					() -> InvalidDataException.extraProperty("deviceOperatingSystem", hint));
```

Comment:
Shouldn't this be `missingProperty` instead of `extraProperty`?

Reply:
You're totally right. It's a copy/paste error. I've pushed an update to correct this.